lib/_tls_wrap.js

```diff
@@ -407,7 +410,12 @@ TLSSocket.prototype._wrapHandle = function(wrap) {
   const context = options.secureContext ||
                   options.credentials ||
                   tls.createSecureContext(options);
-  const res = tls_wrap.wrap(handle._externalStream,
+  const externalStream = handle._externalStream;
+  assert(typeof externalStream === 'object',
+         'handle must be a LibuvStreamWrap');
```

Comment:
It just needs to be a `StreamBase`, even if `LibuvStreamWrap` is the common case here.

Reply:
@addaleax I thought about checking the type of `externalStream` here but there doesn't seem to be a way of checking those in JS land since there isn't a proper inheritance.

Reply:
@joyeecheung Yeah, `externalStream` doesn’t really tell you what kind of thing it is. :/

Ideally, what we could do is making the `StreamBase` inheritance hierarchy reflected in the actual JS handle classes; that requires more work than this PR, though…

Reply:
@addaleax Yeah that should probably better be done in a dedicated PR IMO